src/blocks/header/lotus_json.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+use crate::lotus_json::*;
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+
+use super::BlockHeader;
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct BlockHeaderLotusJson {
+    miner: AddressLotusJson,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    ticket: Option<TicketLotusJson>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    election_proof: Option<ElectionProofLotusJson>,
+    beacon_entries: VecLotusJson<BeaconEntryLotusJson>,
+    win_po_st_proof: VecLotusJson<PoStProofLotusJson>,
+    parents: TipsetKeysLotusJson,
+    parent_weight: BigIntLotusJson,
+    height: i64,
+    parent_state_root: CidLotusJson,
+    parent_message_receipts: CidLotusJson,
+    messages: CidLotusJson,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    b_l_s_aggregate: Option<SignatureLotusJson>,
+    timestamp: u64,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    block_sig: Option<SignatureLotusJson>,
+    fork_signaling: u64,
+    parent_base_fee: TokenAmountLotusJson,
+}
+
+impl HasLotusJson for BlockHeader {
+    type LotusJson = BlockHeaderLotusJson;
+
+    fn snapshots() -> Vec<(serde_json::Value, Self)> {
```

Comment:
What is the purpose of this method?

Reply:
It's for testing our de/serialization: https://jestjs.io/docs/snapshot-testing
This makes sure every struct has a test

Reply:
Shouldn't it be only visible under tests?

Reply:
We _could_ do that, but it would require a second trait, or a `#[cfg(test)]` under the trait definition, both of which require more verbosity for the implementor - I want the fast path to be a quick as possible for users - one trait to implement, and they get tested etc for free

Reply:
For me, it's a bit confusing if I'd see something like this in method suggestions. Let's mark it somehow that it shouldn't be used outside tests.

Reply:
I played around with adding `#[cfg(test)]` for this, but it kills e.g autocomplete - traits are not expected to have `#[cfg(...)]` members.

I'm keeping this as is for ease of use, but I've improved the documentation for this method.